modules/fbs/fbs.module

```diff
@@ -649,31 +649,47 @@ function fbs_form_profile2_phone_validate($element, $form_state) {
 /**
  * On hold validation for fbs_form_profile2
  */
-function fbs_form_profile2_onhold_validate($element, $form_state) {
+function fbs_form_profile2_onhold_validate($element, &$form_state, $form) {
   $from_date = $form_state['input']['profile_provider_fbs']['field_fbs_on_hold']['und'][0]['from']['date'];
   $from_date_stamp = strtotime($from_date);
+  $default_from_date = $form['profile_provider_fbs']['field_fbs_on_hold']['und'][0]['from']['#default_value'];
+  $default_from_date_stamp = strtotime($default_from_date);
   $to_date = $form_state['input']['profile_provider_fbs']['field_fbs_on_hold']['und'][0]['to']['date'];
   $to_date_stamp = strtotime($to_date);
   $now = date('d-m-Y');
   $now_stamp = strtotime($now);
 
+  if (!empty($default_from_date) && 
+    $from_date_stamp != $default_from_date_stamp && 
+    $from_date_stamp <= $now_stamp) {
+    form_set_error('profile_provider_fbs][field_fbs_on_hold', 
+      t('Your reservations is already on hold from start date: %default-date. If you wish to change the start date from: %default-date, the start date must be higher than or equal to today: %now', 
+      array(
+        '%default-date' => format_date($default_from_date_stamp, 'custom', 'd-m-Y'), 
+        '%now' => $now),
+      )
+    );
+  }
+  if (empty($default_from_date) && $from_date_stamp <= $now_stamp) {
+    form_set_error('profile_provider_fbs][field_fbs_on_hold', t('Start date must be higher than or equal to today: %now', array('%now' => $now)));
+  }
   if ($to_date_stamp < $now_stamp && $to_date_stamp < $from_date_stamp) {
     form_set_error('profile_provider_fbs][field_fbs_on_hold', t('To date must both be higher than or equal to today, and be higher than the start date.'));
   }
   elseif ($to_date_stamp < $now_stamp && $to_date_stamp > $from_date_stamp) {
-    form_set_error('profile_provider_fbs][field_fbs_on_hold', t('To date must both be higher than or equal to today') . ' - ' . $now . '.');
+    form_set_error('profile_provider_fbs][field_fbs_on_hold', t('To date must both be higher than or equal to today: %now', array('%now' => $now)));
   }
-  elseif ($to_date_stamp >= $now_stamp && $to_date_stamp <= $from_date_stamp) {
+  elseif ($to_date_stamp > $now_stamp && $to_date_stamp <= $from_date_stamp) {
     form_set_error('profile_provider_fbs][field_fbs_on_hold', t('To date must higher than the from date.'));
   }
 }
 
 /**
  * Submit handler for fbs_form_profile2
  */
-function fbs_form_profile2_rebuild_submit($form, &$form_state) {
-  // Rebuilds form and stay on same page.
-  $form_state['rebuild'] = TRUE;
+function fbs_form_profile2_redirect_submit($form, &$form_state) {
+  // Redirects form to stay on same page on submit.
+  $form_state['redirect'] = $_GET['q'];
```

Comment:
Do we need this submit handler at all if we just want to redirect to the same page? That seems to be the default behavior of [`drupal_redirect_form()`](https://api.drupal.org/api/drupal/includes%21form.inc/function/drupal_redirect_form/7.x)?

Reply:
This form do not redirect to the same page, but redirects to /user/[uid]